packages/foam-vscode/src/core/janitor/generateLinkReferences.test.ts

```diff
@@ -8,19 +8,21 @@ import { URI } from '../model/uri';
 import { FoamWorkspace } from '../model/workspace';
 import { FileDataStore, Matcher } from '../services/datastore';
 import { Logger } from '../utils/log';
+import { getName } from '../utils/path';
 
 Logger.setLevel('error');
 
 describe('generateLinkReferences', () => {
   let _workspace: FoamWorkspace;
+  // TODO slug must be reserved for actual slugs, not file names
   const findBySlug = (slug: string): Resource => {
     return _workspace
       .list()
-      .find(res => URI.getBasename(res.uri) === slug) as Resource;
```

Comment:
Please see my comments below. This is a slippery slope, if I add `getBasename` I should as well add a bunch of other methods. This problem is not apparent now because path manipulation logic is scattered across the codebase, but as soon as I tried to unify all usages in a coherent path API it became clear that there are many functions and having them twice, one as functions over plain strings and one as URI wrappers to those functions, is not very nice. I've suggested a number of alternatives, but I still find the current one desirable since IMO better balances the different concerns here: code duplication, low-level (string) vs high-level path access, etc.